lib/workers/repository/onboarding/branch/index.ts

```diff
@@ -129,3 +121,15 @@ function handleOnboardingManualRebase(onboardingPr: Pr): void {
     OnboardingState.prUpdateRequested = true;
   }
 }
+
+function invalidateExtractCache(baseBranch: string): void {
+  const cache = getCache();
+  cache.scan ||= {};
+  logger.debug(
+    `Onboarding branch modified. Removing outdated extract cache for branch=${baseBranch}`
```

Comment:
Shouldn't it be invalidated if the default branch or any base branch SHA changes too?

Reply:
So then the log message is wrong/misleading some of the time?

Reply:
Don't think so. If: 
1. onboardingBranch is modified we log 
```Onboarding branch modified. Removing outdated extract cache for branch=${baseBranch}```

2.  baseBranch/defaultBranch is updatedwe log 
```Cached extract result cannot be used due to base branch SHA change ...```

3.  baseBranch updated & onboardingBranch modified
```Onboarding branch modified. Removing outdated extract cache for branch=${baseBranch}``` 

4. none of the branches have changed
```Cached extract for sha=${baseBranchSha} is valid and can be used```

Reply:
So the only time `invalidateExtractCache()` is called is for option (1)?

Reply:
yes.

Reply:
The function seems misnamed then. If someone were to call it for other reasons in future then you'd end up with a wrong debug message. Either move the logging out of the function or rename the function